sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorageTest.java

```diff
@@ -120,7 +124,26 @@ void longAsynchronousAccumulator_AttributesProcessor_used() {
   }
 
   @Test
-  void collectAndReset_CatchesRuntimeException() {
+  void collectAndReset_CallbackExceptions() {
+    // Should catch runtime exceptions
+    assertThat(
+            collectAndReset(
+                unused -> {
+                  throw new RuntimeException("Error!");
+                }))
+        .isEqualTo(null);
+
+    // Should propagate fatal exceptions
+    assertThatCode(
```

Comment:
Ditto (or maybe just delete, not sure if it's worth having this test case in every single location we use that helper)

Reply:
Yeah wasn't sure what your thoughts would be on this. Ideally we'd be able to verify that the method was called, but because its static, the only way to really verify it is by duplicating the coverage from `ThrowableUtilTest`.